Apps/Laboratory/src/Services/LaboratoryService.cs

```diff
@@ -86,224 +86,163 @@ public class LaboratoryService : ILaboratoryService
         {
-            RequestResult<Covid19OrderResult> retVal = new()
-            {
-                ResourcePayload = new(),
-                ResultStatus = ResultType.Error,
-                ResultError = UnauthorizedResultError(),
-            };
-
             string? accessToken = this.authenticationDelegate.FetchAuthenticatedUserToken();
-
-            if (accessToken != null)
+            if (accessToken == null)
             {
-                RequestResult<PhsaResult<List<PhsaCovid19Order>>> delegateResult =
-                    await this.laboratoryDelegate.GetCovid19Orders(accessToken, hdid, pageIndex).ConfigureAwait(true);
-
-                retVal.ResultStatus = delegateResult.ResultStatus;
-                retVal.ResultError = delegateResult.ResultError;
-                retVal.PageIndex = delegateResult.PageIndex;
-                retVal.PageSize = delegateResult.PageSize;
-                retVal.TotalResultCount = delegateResult.TotalResultCount;
+                return RequestResultFactory.Error<Covid19OrderResult>(UnauthorizedResultError());
+            }
 
-                if (delegateResult.ResultStatus == ResultType.Success)
-                {
-                    retVal.ResourcePayload.Covid19Orders =
-                        this.autoMapper.Map<IEnumerable<PhsaCovid19Order>, IEnumerable<Covid19Order>>(delegateResult.ResourcePayload?.Result);
-                }
+            RequestResult<PhsaResult<List<PhsaCovid19Order>>> delegateResult = await this.laboratoryDelegate.GetCovid19Orders(accessToken, hdid, pageIndex).ConfigureAwait(true);
 
-                PhsaLoadState? loadState = delegateResult.ResourcePayload?.LoadState;
-                if (loadState != null)
-                {
-                    retVal.ResourcePayload.Loaded = !loadState.RefreshInProgress;
-                    if (loadState.RefreshInProgress)
+            PhsaLoadState? loadState = delegateResult.ResourcePayload?.LoadState;
+            if (loadState != null && loadState.RefreshInProgress)
+            {
+                return RequestResultFactory.ActionRequired(
+                    new Covid19OrderResult
                     {
-                        retVal.ResultStatus = ResultType.ActionRequired;
-                        retVal.ResultError = ErrorTranslator.ActionRequired("Refresh in progress", ActionType.Refresh);
-                        retVal.ResourcePayload.RetryIn = Math.Max(
-                            loadState.BackOffMilliseconds,
-                            this.labConfig.BackOffMilliseconds);
-                    }
-                }
+                        RetryIn = Math.Max(loadState.BackOffMilliseconds, this.labConfig.BackOffMilliseconds),
+                        Loaded = !loadState.RefreshInProgress,
+                    },
+                    ActionType.Refresh,
+                    "Refresh in progress");
             }
```

Comment:
The Loaded property also needs to be set when the request is successful.

Reply:
Doesn't line 104 set the this property?

Reply:
This still needs to be fixed.